support-frontend/assets/components/directDebit/directDebitProgressiveDisclosure/components/form.jsx

```diff
@@ -0,0 +1,126 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { compose } from 'redux';
+
+import Button from 'components/button/button';
+import { ErrorSummary } from 'components/subscriptionCheckouts/submitFormErrorSummary';
+import GeneralErrorMessage
+  from 'components/generalErrorMessage/generalErrorMessage';
+import { Input } from 'components/forms/input';
+import { withLabel } from 'hocs/withLabel';
+import { withError } from 'hocs/withError';
+import { CheckboxInput } from 'components/forms/customFields/checkbox';
+import { type ErrorReason } from 'helpers/errorReasons';
+import { type Option } from 'helpers/types/option';
+
+import '../directDebitForm.scss';
+
+
+const InputWithError = compose(withError, withLabel)(Input);
+const CheckBoxWithError = withError(CheckboxInput);
+
+function Form(props: {
+  accountHolderName: string,
+  sortCodeString: string,
+  accountNumber: string,
+  accountHolderConfirmation: boolean,
+  accountHolderNameError: string,
+  sortCodeError: string,
+  accountNumberError: string,
+  accountHolderConfirmationError: string,
+  showGeneralError: boolean,
+  accountErrorsLength: number,
+  accountErrors: Array<Object>,
+  submissionError: ErrorReason | null,
+  submissionErrorHeading: string,
+  allErrors: Array<Object>,
+  formError: Option<string>,
+  updateAccountHolderName: (accountHolderName: SyntheticInputEvent<HTMLInputElement>) => void,
```

Comment:
If you define a type `type EventHandler = (e: SyntheticInputEvent<HTMLInputElement>) => void;` this can be
```suggestion
  updateAccountHolderName: EventHandler,
  updateSortCodeString: EventHandler,
  updateAccountNumber: EventHandler,
  updateAccountHolderConfirmation: EventHandler,
```